recipes/uvw/all/conanfile.py

```diff
@@ -13,73 +18,56 @@ class UvwConan(ConanFile):
     license = "MIT"
     url = "https://github.com/conan-io/conan-center-index"
     homepage = "https://github.com/skypjack/uvw"
+    package_type = "header-library"
     no_copy_source = True
     settings = "compiler"
 
     @property
-    def _source_subfolder(self):
-        return "source_subfolder"
+    def _min_cppstd(self):
+        return 17
 
     @property
-    def _supported_compiler(self):
-        compiler = str(self.settings.compiler)
-        version = tools.Version(self.settings.compiler.version)
-        if compiler == "Visual Studio" and version >= "15":
-            return True
-        if compiler == "gcc" and version >= "7":
-            return True
-        if compiler == "clang" and version >= "5":
-            return True
-        if compiler == "apple-clang" and version >= "10":
-            return True
-        return False
-
-    def configure(self):
-        if self.settings.compiler.get_safe("cppstd"):
-            tools.check_min_cppstd(self, "17")
-        if not self._supported_compiler:
-            raise ConanInvalidConfiguration("uvw requires C++17. {} {} does not support it.".format(
-                str(self.settings.compiler),
-                self.settings.compiler.version)
-            )
+    def _compilers_minimum_version(self):
+        return {
+            "Visual Studio": "15",
+            "gcc": "7",
+            "clang": "5",
+            "apple-clang": "10",
+        }
 
     @property
-    def _required_EXACT_libuv_version(self):
-        """ Return *EXACT* libuv version to use with this uvw library """
+    def _required_libuv_version(self):
+        # uvw is bound to a particular libuv version which is part of the release archive name
         match = re.match(r".*libuv[_-]v([0-9]+\.[0-9]+).*", self.conan_data["sources"][self.version]["url"])
         if not match:
-            raise ConanException("uvw recipe does not know what version of libuv to use as dependency")
-        return tools.Version(match.group(1))
+            raise ConanException(f"{self.ref} does not know what version of libuv to use as a dependency")
+        return match.group(1)
+
+    def layout(self):
+        basic_layout(self, src_folder="src")
 
     def requirements(self):
-        libuv_version = self._required_EXACT_libuv_version
-        revision = 0
-        if libuv_version.major == "1" and libuv_version.minor == "44":
-            revision = 1
-        self.requires("libuv/{}.{}.{}".format(libuv_version.major, libuv_version.minor, revision))
+        self.requires(f"libuv/[~{self._required_libuv_version}]")
```

Comment:
I believe the previous version check was incorrect. uvw releases are tight to libuv releases but only the major and minor version need to match as per [uvw versioning scheme](https://github.com/skypjack/uvw#versioning).

Reply:
Need to used a specific version. Version is only supported by using OpenSSL and CMake for now: https://github.com/conan-io/conan-center-index/blob/master/docs/adding_packages/dependencies.md#version-ranges. Please, update to use a single version.

Reply:
OK. Thank you. I will fix it.

Reply:
uvw depends on `major.minor` version of libuv. I would like the formula to always use the latest libuv at least at the time of updating uvw. Therefore I added `uvw_version` into conandata.

Please, let me know if it's acceptable or if there is another preferred way @uilianries.

I could always move it into some dictionary within the conanfile.

Reply:
Let us think about how best handle it and we'll come back to you after we discuss it tomorrow :)